vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -118,7 +169,10 @@ def _transformed_inputs(self) -> Union[list[State], dict[str, Union[State, Contr
         """
         if self._legacy:
             # Must be an Action rather than _AbstractAction, so has already been validated by pydantic field annotation.
-            return [State(*input.split(".")) for input in cast(Action, self).inputs]
+            return [
+                State(*self._transform_dependency(input, type="input").split("."))
+                for input in cast(Action, self).inputs
+            ]
```

Comment:
Unless you have strong feelings otherwise, let's revert this so that the new behaviour is only possible for non-legacy actions.

Reply:
ok, let's keep it in then (both the transform and using the transformed values). I was just thinking it might be clearer to make a clean split between legacy actions where no new features are possible vs. new actions where all the new features are possible. But here it's easy to keep the new feature in and you get the benefit of validation which we'd be losing otherwise for the legacy actions. So leave as you had it.